src/Highlight.ts

```diff
@@ -91,17 +100,48 @@ export default class Highlight {
   }
 
   public serialize(referenceElement?: HTMLElement): SerializedHighlight {
-    referenceElement = referenceElement || dom(this.range.commonAncestorContainer).closest('[id]');
+    const validReferenceElement = this.getValidReferenceElement(
+      this.options.skipIDsBy, referenceElement
```

Comment:
`getValidReferenceElement` is a class method that has access to `this.options`, so it seems unnecessary to pass it in